send/ADConnector.pm

```diff
@@ -118,6 +127,59 @@ sub init_config($) {
 
 }
 
+#
+# Resolve available domain controlers from a URL
+#
+sub resolve_domain_controlers($) {
+	my $ldap_location = shift;
+	my ($protocol, $host, $port);
+	if($ldap_location =~ /^([^\/]+:\/\/)?([^:]+):([0-9]+)$/) {
+		($protocol, $host, $port) = ($1, $2, $3);
+	} else {
+		ldap_log('ad_connection', "[AD] Ldap location in wrong format. Expected: protocol://host:port");
+		print STDERR "[AD] Ldap location in wrong format. Expected: protocol://host:port";
+		die "[AD] Ldap location in wrong format. Expected: protocol://host:port";
+	}
+	my $resolver = Net::DNS::Resolver->new;
+	my $query = $resolver->search($host);
+	my @controlers = ();
+
+	if ($query) {
+  		foreach ($query->answer) {
+    		next unless $_->type eq "A";
+			push(@controlers, $protocol . $_->address . ":" . $port);
+  		}
+	} else {
+		my $error = $resolver->errorstring();
+		ldap_log('ad_connection', "[AD] No answer was found for $host. DNS query returned error message: $error");
+        print STDERR "[AD] No answer was found for $host. DNS query returned error message: $error";
+        die "[AD] No answer was found for $host. DNS query returned error message: $error";
```

Comment:
Indent ;-)

Also, print to STDERR is not necessary because that's already covered by die.

Reply:
Fixed